chemprop/uncertainty/evaluator.py

```diff
@@ -68,8 +82,12 @@ def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -
 @UncertaintyEvaluatorRegistry.register("spearman")
 class SpearmanEvaluator(UncertaintyEvaluator):
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        ...
-        return
+        masked_preds = preds * mask
+        masked_targets = targets * mask
+        masked_uncs = uncs * mask
+        masked_errs = (masked_preds - masked_targets).abs()
+        spearman = SpearmanCorrCoef(num_outputs=masked_targets.shape[1])
```

Comment:
```suggestion
        masked_uncs = uncs[mask]
        masked_errs = (preds[mask] - targets[mask]).abs()
        spearman = SpearmanCorrCoef(num_outputs=mask.shape[1])
```
`preds * mask` will make the returned value being 0 for the corresponding mask value being False. I think we should ignore those cases.

```
import torch

preds = torch.tensor([[1, 2, 3], [4, 5, 6]])
mask = torch.tensor([[True, False, True], [False, True, False]])

# preds * mask
result1 = preds * mask
# tensor([[1, 0, 3],
#         [0, 5, 0]])

# preds[mask]
result2 = preds[mask]
# tensor([1, 3, 5])
```

Reply:
Agreed. As we discussed, doing `preds[mask]` and the like would flatten them which is not quite what we want as we want to retain the dimension for the number of targets. I will follow Nathan's approach with the Platt calibrator as suggested.